added command line argument for csv / json. csv is default

joshkenney opened the pull request.

joshkenney committed:
added json output to songs and podcasts

joshkenney committed:
added json to .gitignore

joshkenney committed:
improved demo in README.md to include cargo build

joshkenney committed:
Merge branch 'raleighlittles:main' into main

joshkenney committed:
Update README.md

joshkenney committed:
added command line argument for csv (default) json

joshkenney committed:
standardized output to say if saving csv or json

raleighlittles commented on Cargo.toml:
```diff
@@ -0,0 +1,6 @@
+[package]
```
There's already a Cargo file in the `parser` directory, we don't need another one here. Delete this file

raleighlittles commented on .gitignore:
```diff
@@ -1,4 +1,11 @@
+repopack-output.txt
+
+
+# Added by cargo
```
the target and debug folders are already in the `.gitignore` in the parser directory. Should not need to modify this file

raleighlittles replied:
The .gitignore does not need any changes

raleighlittles commented on README.md:
````diff
@@ -82,7 +82,10 @@ The 7 possible "type" options are:
 
 ```bash
-$ /target/debug/itunesdb_parser <path-to-itunesdb-file> '<type>'
+$ cd iTunesDB-Parser
````
1. the path to build in is the `parser` directory because thats where the code is
2. No need to run `cargo init` that is for new projects

raleighlittles replied:
This path is already part of what is cloned, there's no need to include it here. You only need to change directory into `parser` to do the build

raleighlittles replied:
so it should be cd into parser, then cargo build, then run the target binary

joshkenney committed:
Merge branch 'main' into main

joshkenney committed:
responding to comments

joshkenney committed:
Merge branch 'main' of github.com:joshkenney/itunes-db-parser

joshkenney commented:
@raleighlittles thanks - deleted upper level Config.toml and Cargo.lock

re: README
i need to use this path 
```
$ cd iTunesDB-Parser
$ ./parser/target/debug/itunesdb_parser <path-to-itunesdb-file> '<type>'
```
as written yours does nothing
```
➜  itunes-db-parser git:(main) ✗ target/debug/itunesdb_parser ~/Downloads/GEOFFREY/iPod_Control/iTunes/iTunesDB "itunes" 
zsh: no such file or directory: target/debug/itunesdb_parser
```

joshkenney committed:
Merge branch 'main' into main

raleighlittles commented on src/main.rs:
```diff
@@ -0,0 +1,3 @@
+fn main() {
```
Delete this file

joshkenney replied:
not sure how this got there, thanks!

joshkenney committed:
removed src/main.rs

joshkenney committed:
updated README to have the proper sequence for build and running the target

raleighlittles commented:
Can you post a JSON file that your branch creates? Run the parser on one of the sample itunesDB files - let me know which. Also post a CSV output from the same iTunesDB file so I can compare it against dev. If it matches I would be ready to merge you. @joshkenney

joshkenney commented:
@raleighlittles done!
`./parser/target/debug/itunesdb_parser ./sample-files/2023-06-26_iTunesDB-1 itunes csv`
`./parser/target/debug/itunesdb_parser ./sample-files/2023-06-26_iTunesDB-1 itunes json`
[music.json](https://github.com/user-attachments/files/17984907/music.json)
[music.csv](https://github.com/user-attachments/files/17984908/music.csv)

joshkenney committed:
improved create_playlist_from_tracks() to batch 100 songs
this respects the limit set by Spotify API andallows for the sumbissions of playlists greater than 100 songs

joshkenney committed:
improved create_playlist_from_tracks() to batch 100 songs
this respects the limit set by Spotify API andallows for the sumbissions of playlists greater than 100 songs

raleighlittles commented:
Looks good to me.
